event_model/__init__.py

```diff
@@ -499,6 +499,8 @@ def compose_resource(*, start, spec, root, resource_path, resource_kwargs,
                      path_semantics=os.name, uid=None, validate=True):
     if uid is None:
         uid = str(uuid.uuid4())
+    if path_semantics == 'nt':
```

Comment:
I worry that this might confuse users who look at the function signature (i.e. `compose_resource?` in IPython) and find a default valid that is not a valid value. Have you considered doing this translation from `os.name` to `path_semantics` above the function definition rather than inside it?

```py
default_path_semantics = {...}[os.name]


def compose_resource(..., path_semantics=default_path_semantics, ...)
```

Reply:
updated